packages/create-fuels/src/cli.ts

```diff
@@ -61,19 +98,55 @@ export const runScaffoldCli = async (
     packageManager = packageManagerInput.packageManager;
   }
 
+  if (!explicitProgramsToInclude) {
+    const programsToIncludeInput = await prompts({
+      type: 'multiselect',
+      name: 'programsToInclude',
+      message: 'Which Sway programs do you want?',
+      choices: [
+        { title: 'Contract', value: 'contract', selected: true },
+        { title: 'Predicate', value: 'predicate' },
+        { title: 'Script', value: 'script' },
+      ],
+      instructions: false,
+    });
+    programsToInclude = {
+      contract: programsToIncludeInput.programsToInclude.includes('contract'),
+      predicate: programsToIncludeInput.programsToInclude.includes('predicate'),
+      script: programsToIncludeInput.programsToInclude.includes('script'),
+    };
+  }
+
   await mkdir(projectPath);
 
   await cp(join(__dirname, '../templates/nextjs'), projectPath, { recursive: true });
   await rename(join(projectPath, 'gitignore'), join(projectPath, '.gitignore'));
 
+  // delete the programs that are not to be included
+  if (!programsToInclude.contract) {
+    rmSync(join(projectPath, 'sway-programs/contract'), { recursive: true });
+  }
+  if (!programsToInclude.predicate) {
+    rmSync(join(projectPath, 'sway-programs/predicate'), { recursive: true });
+  }
+  if (!programsToInclude.script) {
+    rmSync(join(projectPath, 'sway-programs/script'), { recursive: true });
+  }
```

Comment:
```suggestion
  await cp(join(__dirname, '../templates/nextjs'), projectPath, {
    recursive: true,
    filter: (_source, destination) => {
      switch (destination) {
        case join(projectPath, 'sway-programs/contract'):
          return programsToInclude.contract;
        case join(projectPath, 'sway-programs/script'):
          return programsToInclude.script;
        case join(projectPath, 'sway-programs/predicate'):
          return programsToInclude.predicate;
        default:
          return true;
      }
    },
  });
  await rename(join(projectPath, 'gitignore'), join(projectPath, '.gitignore'));
```

You can skip copying the folders instead of copying and then deleting them. Note that I didn't try out the code above. I still wanted to create a suggestion for the purpose of showcasing the `filter` in the full context.

Reply:
@nedsalk let me explain the rationale here: I am selectively deleting the programs I don't need because the 'template' has to have all three programs since we copy the _whole_ template folder at once. And I don't think it is a good idea to have three different templates for the three different program types. The current design also easily enables the users to choose multiple program types.